blivet/deviceaction.py

```diff
@@ -690,8 +690,11 @@ def apply(self):
     def execute(self, callbacks=None):
         """ wipe the filesystem signature from the device """
         # remove any flag if set
-        if self.format.parted_flag:
-            self.device.unset_flag(self.format.parted_flag)
+        if isinstance(self.device, PartitionDevice) and self.device.disklabel_supported:
+            if self.format.parted_flag:
+                self.device.unset_flag(self.format.parted_flag)
+            if self.format.parted_system is not None:
+                self.device.parted_partition.system = None
```

Comment:
Don't you need to also do `self.device.disk.commit_to_disk()` or similar, as in `ActionCreateFormat.execute`?